trace/es/elasticsearch_test.go

```diff
@@ -0,0 +1,178 @@
+	assert.EqualValues(t, hostsFmt, actualTags["hosts"])
+
+	assert.Equal(t, "dev", actualTags["version"])
+	assert.Equal(t, "go-elasticsearch", actualTags["component"])
+	assert.Equal(t, "elasticsearch", actualTags["db.type"])
+	assert.Equal(t, user, actualTags["db.user"])
+	assert.Equal(t, "/", actualTags["http.url"])
+	assert.Equal(t, method, actualTags["http.method"])
+
+	respondent := "es.respondent.com:9200"
+	statusCode := 200
+	rsp := &http.Response{
+		Request: &http.Request{
+			URL: &url.URL{
+				Host: respondent,
+			},
+		},
+		StatusCode: statusCode,
+	}
+	endSpan(sp, rsp)
+
+	// The end of span should append the below three tags
+	jsp = sp.(*mocktracer.MockSpan)
+	assert.Equal(t, respondent, jsp.Tag(respondentTag))
+	assert.Equal(t, uint16(statusCode), jsp.Tag("http.status_code"))
+	assert.Equal(t, false, jsp.Tag("error"))
+
+	// Anything else should remain exactly the same
+	actualResponseTags := jsp.Tags()
+	delete(actualResponseTags, "http.status_code")
+	delete(actualResponseTags, respondentTag)
+	delete(actualResponseTags, "error")
+	assert.EqualValues(t, actualTags, actualResponseTags)
+}
+
+func TestNewDefaultClient(t *testing.T) {
+	newClient := NewDefaultClient()
+	upstreamClient, err := elasticsearch.NewDefaultClient()
+	assert.Nil(t, err)
+	assert.IsType(t, *upstreamClient, newClient.Client)
+
+	expectedTransport, transport := new(transportClient), newClient.Transport
+	assert.IsType(t, expectedTransport, transport)
+
+	defaultAddr := getAddrFromEnv()
+	expectedURL, err := url.Parse(strings.TrimRight(defaultAddr, "/"))
+	assert.Nil(t, err)
+	cfg := estransport.Config{
+		URLs:      []*url.URL{expectedURL},
+		Transport: nil,
+	}
+	expectedTransport.client = estransport.New(cfg)
+	expectedTransport.tracingInfo.hosts = []string{defaultAddr}
+	assert.EqualValues(t, expectedTransport, transport)
+}
+
+func TestNewClient(t *testing.T) {
+	addresses := []string{"http://www.host1.com:9200", "https://10.1.1.1:9300"}
+	user, password, apiKey := "user1", "pass", "key"
+	cfg := Config{
+		Addresses: addresses,
+		Username:  user,
+		Password:  password,
+		APIKey:    apiKey,
+	}
+
+	newClient, err := NewClient(cfg)
+	assert.Nil(t, err)
+	assert.IsType(t, new(Client), newClient)
+
+	expectedTransport, transport := new(transportClient), newClient.Transport
+	assert.IsType(t, expectedTransport, transport)
+
+	expectedURLs, err := addrsToURLs(addresses)
+	assert.Nil(t, err)
+	transportCfg := estransport.Config{
+		URLs:      expectedURLs,
+		Username:  user,
+		Password:  password,
+		APIKey:    apiKey,
+		Transport: nil,
+		Logger:    nil,
+	}
+	expectedTransport.client = estransport.New(transportCfg)
+	expectedTransport.tracingInfo.hosts = addresses
+	expectedTransport.user = user
+	assert.EqualValues(t, expectedTransport, transport)
+}
+
+func TestEsQuery(t *testing.T) {
+	mtr := mocktracer.New()
+	opentracing.SetGlobalTracer(mtr)
+	assert.Empty(t, mtr.FinishedSpans())
+
+	httpmock.Activate()
```

Comment:
We can use HTTP testing with std `httptest` and avoid depending on one more package.